iijpta/ngx_http_secure_token_iijpta.c

```diff
@@ -0,0 +1,231 @@
+#include <ngx_config.h>
+
+#include "ngx_http_secure_token_iijpta.h"
+#include "../ngx_http_secure_token_filter_module.h"
+#include "../ngx_http_secure_token_utils.h"
+
+#include <openssl/evp.h>
+
+// constants
+#define CRC32_SIZE    4
+#define EXPIRY_SIZE   8
+#define PATH_LIMIT    1024
+
+static char *ngx_conf_check_byte_len_bounds(ngx_conf_t *cf, void *post, void *data);
+
+// typedefs
+typedef struct {
+	ngx_str_t key;
+	ngx_str_t iv;
+	ngx_str_t path;
+	ngx_secure_token_time_t end;
+} ngx_secure_token_iijpta_token_t;
+
+typedef struct {
+        u_char crc[CRC32_SIZE];
+        u_char expiry[EXPIRY_SIZE];
+} ngx_http_secure_token_iijpta_header_t;
+
+static ngx_conf_num_bounds_t ngx_http_secure_token_iijpta_key_bounds = {
+	ngx_conf_check_byte_len_bounds, 16, 16
+};
+
+static ngx_conf_num_bounds_t ngx_http_secure_token_iijpta_iv_bounds = {
+	ngx_conf_check_byte_len_bounds, 16, 16
+};
+
+// globals
+static ngx_command_t ngx_http_secure_token_iijpta_cmds[] = {
+	{ ngx_string("key"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, key),
+	&ngx_http_secure_token_iijpta_key_bounds },
+
+	{ ngx_string("iv"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, iv),
+	&ngx_http_secure_token_iijpta_iv_bounds },
+
+	{ ngx_string("path"),
```

Comment:
Better rename to acl for consistency with other tokens + change to 'complex value' in order to support variables, such as $uri (see Akamai as an example)